lib/Db/CircleRequest.php

```diff
@@ -202,6 +203,48 @@ public function getCircles(?IFederatedUser $initiator, CircleProbe $probe): arra
 	}
 
 
+	/**
+	 * get data about multiple Circles.
+	 *
+	 * - CircleProbe is used to define the list of circles to be returned by the method,
+	 * - DataProbe is used to define the complexity of the data to be returned for each entry of the list
+	 *
+	 * @param IFederatedUser|null $initiator
+	 * @param CircleProbe $circleProbe
+	 * @param DataProbe $dataProbe
+	 *
+	 * @return array
+	 * @throws RequestBuilderException
+	 */
+	public function probeCircles(
+		?IFederatedUser $initiator,
+		CircleProbe $circleProbe,
+		DataProbe $dataProbe
+	): array {
+		$qb = $this->getCircleSelectSql();
+		if (!$dataProbe->has(CoreQueryBuilder::MEMBERSHIPS)) {
+			$dataProbe->add(CoreQueryBuilder::MEMBERSHIPS);
+		}
+
+		$qb->setAlternateSqlPath(CoreQueryBuilder::CIRCLE, $dataProbe->getPath())
```

Comment:
This "alternate" naming is confusing, this should be setSqlPath
It has a default value and you can set it to change it.